ENG-2633 - Clarify password required (#3713)
FusionAuth/fusionauth-site: astro/src/content/docs/apis/_user-request-body.mdx

    { props.http_method === 'POST' &&
        <p>

>           If <code>tenant.passwordEnabled</code> is true, this field is required unless <code>sendSetPasswordEmail</code> or  <code>sendSetPasswordPhone</code> is set to <code>true</code>. By default, <code>sendSetPasswordEmail</code> and <code>sendSetPasswordPhone</code> are <code>false</code>.

Reviewer: ```suggestion
          This field is required unless `tenant.passwordEnabled` is `false`, or either `sendSetPasswordEmail` or `sendSetPasswordPhone` is `true`. Note that by default, `sendSetPasswordEmail` and `sendSetPasswordPhone` are `false`.
```

Author: See https://github.com/FusionAuth/fusionauth-site/pull/3713#discussion_r2089639921
